src/js/editor/editor.js

```diff
@@ -305,13 +305,6 @@ class Editor {
   }
 
   reparse() {
-    let { headSection, headSectionOffset } = this.cursor.offsets;
-    if (headSectionOffset === 0) {
-      // FIXME if the offset is 0, the user is typing the first character
-      // in an empty section, so we need to move the cursor 1 letter forward
-      headSectionOffset = 1;
-    }
```

Comment:
The only thing that I know of that this was still critical for is handling the OS X "Document shortcut" Control-K (see [apple's docs](https://support.apple.com/en-us/HT201236) for more info) that clears to end of line. Without manually replacing the cursor, it goes shooting off somewhere crazy.

We should just reopen #96 and add a key command for control-K, though.